streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -917,8 +922,34 @@ public void punctuate(final ProcessorNode<?, ?, ?, ?> node,
             maybeMeasureLatency(() -> node.punctuate(timestamp, punctuator), time, punctuateLatencySensor);
         } catch (final StreamsException e) {
             throw e;
-        } catch (final RuntimeException e) {
-            throw new StreamsException(String.format("%sException caught while punctuating processor '%s'", logPrefix, node.name()), e);
+        } catch (final Exception e) {
+            final ErrorHandlerContext errorHandlerContext = new DefaultErrorHandlerContext(
+                null,
+                recordContext.topic(),
+                recordContext.partition(),
+                recordContext.offset(),
+                recordContext.headers(),
+                null,
+                null
+            );
+
+            final ProcessingExceptionHandler.ProcessingHandlerResponse response;
+
+            try {
+                response = processingExceptionHandler.handle(errorHandlerContext, null, e);
+            } catch (final Exception fatalUserException) {
```

Comment:
Same here. Why not `RuntimeException` -- given that no checked exception is declared for the `handler(...)` method, it's not possible to get a `Exception` here.

Reply:
@mjsax Same answer: https://github.com/apache/kafka/pull/16300#discussion_r1698085150. `ProcessorNode#process` and `StreamTask#punctuate` stick with `RecordCollectorImpl` and `RecordDeserializer`.